### ISSUE TITLE
Verify setupPush is fast enough after it is moved to lazily initialization

### ISSUE THREAD

**Issue by mcomella** [mozilla-mobile/fenix#8058]
Blocked on #6730.

We're trying to move everything non-essential until after visual completeness in order to improve startup time. Jonathan is working on making setupPush lazily initialize in https://github.com/mozilla-mobile/fenix/issues/6730. We should 1) understand what that change is realing doing and 2) ensure that could will land with enough time for us to mitigate if there are issues.

After that code lands, we should ensure it's fast enough. If not, we should try to move it until after visual completeness, working with (or delegating to) Jonathan to do so.

**Comment by mcomella** [comment]
This will likely come up again in a different context when we look at startup holistically so we'll open a new issue for that: closing.
